flagsmith-engine/index.ts

```diff
@@ -0,0 +1,97 @@
+import { EnvironmentModel } from './environments/models';
+import { FeatureStateModel } from './features/models';
+import { IdentityModel } from './identities/models';
+import { TraitModel } from './identities/traits/models';
+import { getIdentitySegments } from './segments/evaluators';
+import { SegmentModel } from './segments/models';
+import { FeatureStateNotFound } from './utils/errors';
+
+function getIdentityFeatureStatesDict(
+    environment: EnvironmentModel,
+    identity: IdentityModel,
+    overrideTraits?: TraitModel[]
+) {
+    // Get feature states from the environment
+    const featureStates: { [key: number]: FeatureStateModel } = {};
+    for (const fs of environment.featureStates) {
+        featureStates[fs.feature.id] = fs;
+    }
+
+    // Override with any feature states defined by matching segments
+    const identitySegments: SegmentModel[] = getIdentitySegments(
+        environment,
+        identity,
+        overrideTraits
+    );
+    for (const matchingSegment of identitySegments) {
+        for (const featureState of matchingSegment.featureStates) {
+            if (featureStates[featureState.feature.id]) {
+                if (featureStates[featureState.feature.id].isHigherSegmentPriority(featureState)) {
+                    continue;
+                }
+            }
+            // note that feature states are stored on the segment in descending priority
```

Comment:
This comment can be removed as it was incorrect 🤦 hence the need for this change to add the feature segment model.